packages/core/src/httpClient/basicHttpClient.ts

```diff
@@ -215,7 +215,7 @@ export class BasicHttpClient {
   protected async request<ResponseType>(request: HttpRequest) {
     const mutatedRequest = await this.preRequest(request);
     const rawResponse = await this.rawRequest<ResponseType>(mutatedRequest);
-    return this.postRequest(rawResponse, request);
+    return this.postRequest(rawResponse, mutatedRequest);
```

Comment:
One of the two main goals for this PR is comparing new token with the one we used in request. All of the previous implementations were around it, but they were either using response headers (response doesn't have to have authorization header) or using a global variable (race conditions).

Reply:
I don't know why we weren't passing `mutatedRequest` to `this.postRequest` before but to me, it looks fine